### PR TITLE
feat: Modal manager

### PR DESCRIPTION
**Pull request by corn-potage**
- adds modal manager
- modal components were modified to support save, cancel functionality
- decouples open node dir dialog and update node dir
- automatically saves new settings keys with default values from client configs when node settings is opened

known bugs:
- node command not updating after each input change (low priority)

Would like to revisit, to optimize, and clean some of the hacky methods I used in NodeSettingsWrapper, which should also fix the node command update.

### CODE REVIEW

File: src/main/main.ts [hunk 1]
```diff
@@ -88,13 +88,15 @@ const createWindow = async () => {
...
     titleBarStyle: 'hiddenInset',
+
     show: false,
     minWidth: 980,
     minHeight: 480,
     width: 1200,
     height: 820,
     icon: getAssetPath('icon.png'),
     webPreferences: {
+      enableBlinkFeatures: 'CSSColorSchemeUARendering',
```
Comment (jgresham) [inline src/main/main.ts:99]: Does this fix scroll bars in dark mode?

File: src/renderer/Presentational/AddNodeStepper/AddNodeStepperModal.tsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
+// This component could be made into a Generic "FullScreenStepper" component
+// Just make sure to always render each child so that children component state isn't cleard
```
Comment (jgresham) [inline src/renderer/Presentational/AddNodeStepper/AddNodeStepperModal.tsx:2]: +1 for comments :)

File: src/renderer/Presentational/ModalManager/ModalManager.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
+import { useSelector } from 'react-redux';
```
Comment (jgresham) [inline src/renderer/Presentational/ModalManager/ModalManager.tsx:1]: Very clean!

File: src/renderer/Presentational/ModalManager/modalUtils.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  nodeSettings: 'nodeSettings',
+  preferences: 'preferences',
+  addValidator: 'addValidator',
+  clientVersions: 'clientVersions',
+  stopNode: 'stopNode',
+  removeNode: 'removeNode',
+  updateUnavailable: 'updateUnavailable',
+});
+
+export const modalOnChangeConfig = async (
```
Comment (jgresham) [inline src/renderer/Presentational/ModalManager/modalUtils.tsx:30]: Could you add a function description comment for this? It seems key to adding a new modal. Let's say a description geared for me, if I wanted to add a new modal

File: src/renderer/state/modal.ts [hunk 5]
```diff
@@ -0,0 +1,49 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import type { RootState } from './store';
+
+export interface ModalScreen {
+  route: string | undefined;
+  type: 'alert' | 'modal' | undefined;
+}
+
+export interface ModalConfig {
+  preferences: object;
```
Comment (jgresham) [inline src/renderer/state/modal.ts:10]: Should these be optional params? thinking modalConfig for nodeSettings wouldn't want to set the other values

**Review by jgresham — COMMENTED** [review]
remaining talking points:
- [x] Node requirements back into add node modal style
- [ ] A few comments I left

And:
Since last review the changes to MM and modal make them much more decoupled from the content within the modals and cleaner. And overall, this are massive changes that required knowing how things work from end-to-end. Great job on getting to know how the app works and having a good control over it! Congrats! There is a lot going on!

File: src/renderer/state/modal.ts (hunk 5 above)
Comment (corn-potage) [reply src/renderer/state/modal.ts:10]: Oohh good find, this is some old code and we have ModalConfig interface in modalUtils so this isn't necessary. will delete

**Comment by corn-potage** [comment]
@jgresham Thanks!! I just added Node Reqs back into modals, and to match the spec provided, I added isDockerRunning check to determine whether to start the node on Node Req page, or move the user to the Docker installation step. Works perfectly when you turn on/off Docker on the 2nd step! Ready to be reviewed again and merged

**Pull request merged**